src/main.cpp

```diff
@@ -866,12 +928,59 @@ bool CheckTransactionWithoutProofVerification(const CTransaction& tx, CValidatio
 {
     // Basic checks that don't depend on any context
 
-    // Check transaction version
-    if (tx.nVersion < MIN_TX_VERSION) {
+    // Pre-Overwinter accepts version >=1 transaction. Overwinter only accepts version 3 transactions.
+    // To enable soft-forks of the transaction format, define a range of versions which can be accepted.
+    if (tx.fOverwintered) {
+        if (tx.nVersion < OVERWINTER_MIN_TX_VERSION) {
+            return state.DoS(100, error("CheckTransaction(): overwinter version too low"),
+                REJECT_INVALID, "bad-tx-overwinter-version-too-low");
+        }
+        else if (tx.nVersion < CTransaction::OVERWINTER_MIN_CURRENT_VERSION) {
```

Comment:
This check shouldn't be in here, as it's a standard rule check which should only be enforced on the mempool, whereas this function is also enforced on the blockchain. If someone duplicated this code to try and create a soft-forking transaction version, this check would eliminate the soft-forkable range between `OVERWINTER_MIN_TX_VERSION` and `OVERWINTER_MIN_CURRENT_VERSION` (which I don't expect would be usable if we follow convention of incrementing the version number, but could conceivably be).